### PR TITLE
refactor: make services more flexible

### PR DESCRIPTION
**Pull request by dbluhm**
By passing in all dependencies to each method that requires it rather than capturing it in the instance state. This should better support more use cases than the previous structure.

This is a backwards compatible change; the more flexible services are wrapped by the `DIDCommMessaging` class to behave exactly the same way as before. To use the new, more flexible services, use `DIDCommMessagingService` class.

### CODE REVIEW

**Review by mepeltier — APPROVED** [review]
LGTM, though I was involved with the changes so I'll let Colton review before merging

**Review by TheTechmage — APPROVED** [review]
I don't see any glaring issues, plus, this works on the Raspberry Pi demo that I gave at IIW as a drop-in update with no changes to my code. LGTM

**Pull request merged**
